lib/CLI.js

```diff
@@ -1554,85 +1554,91 @@ CLI.monit = function(cb) {
 /**
  * Description
  * @method streamLogs
- * @param {} id
+ * @param {String} id
+ * @param {Number} lines
+ * @param {Boolean} raw
  * @return
  */
+
 CLI.streamLogs = function(id, lines, raw) {
-  if (typeof lines !== 'number')
-    lines = 20;
+  var files_list = [];
+
+  // If no argument is given, we stream logs for all running apps
+  id = id || 'all';
+  lines = lines !== undefined ? lines : 20;
+  lines = lines < 0 ? -(lines) : lines;
+
+  // Avoid duplicates and check if path is different from '/dev/null'
+  var pushIfUnique = function(entry) {
+    var exists = false;
+
+    if (entry.path.toLowerCase
+       && entry.path.toLowerCase() !== '/dev/null') {
+
+      files_list.some(function(file) {
+        if (file.path === entry.path)
+          exists = true;
+        return exists;
+      });
+
+      if (exists)
+        return;
+
+      files_list.push(entry);
+    }
+  };
+
+  // Get the list of all running apps
   Satan.executeRemote('getMonitorData', {}, function(err, list) {
     if (err) {
       printError(err);
       exitCli(cst.ERROR_EXIT);
     }
 
-    if (!raw) {
-      Log.stream(cst.PM2_LOG_FILE_PATH, lines);
-      printOut('[PM2] Starting streaming logs for [%s] process', id || 'all');
-    }
+    if (!raw)
+      printOut('[PM2] Tailing last %d lines for [%s] process%s', lines, id, id === 'all' ? 'es' : '', '\n');
 
-    if (!list || list.length == 0) {
-      return printError('Can not find log files, try to reload logs by executing "pm2 reload <name|all>".');
-    }
-
-    // Group by logs.
-    var groups = {};
-    list.forEach(function(proc){
-      var pk;
-      ['', 'out', 'err'].some(function(n){
-        return ((pk = 'pm_' + (n ? n + '_' : '') + 'log_path') in proc.pm2_env);
-      });
-      pk = proc.pm2_env[pk];
-
-      // Ignored meaningless logs.
-      if (!pk || pk.toLowerCase() == '/dev/null') {
-        return;
+    // Populate the array `files_list` with the paths of all files we need to tail
+    list.forEach(function(proc) {
+      if (proc.pm2_env
+         && (id === 'all'
+            || proc.pm2_env.name == id
+            || proc.pm2_env.pm_id == id)) {
+        if (proc.pm2_env.pm_out_log_path)
```

Comment:
AFAIC, I don't like dry codes :)
It's difficult to maintain copies.
How about do it in this way:
```javascript
['out', 'err'].forEach(function(lt){
  var _path;
  if(_path = proc.pm2_env['pm_' + lt + '_log_path']){
    pushIfUnique({
      path     : _path,
      app_name : proc.pm2_env.name + '-' + proc.pm2_env.pm_id,
      type     : lt
    });
  }
});
```